File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
let [[n, m], ...input] = fs
  .readFileSync(filePath)
  .toString()
  .split("\n")
  .map((x) => x.split(" ").map(Number));

let arr = Array(n)
  .fill()
  .map((_, i) => i + 1);
for (let [i, j] of input) {
  let newArr = [];
  for (let idx = i - 1; idx < j; idx++) {
    newArr.push(arr[idx]);
  }
  newArr.reverse();
  arr.splice(i - 1, j - i + 1, ...newArr);
}
console.log(arr.join(" "));

File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.py
N, M = map(int, input().split())

basket = [ n for n in range(0,N + 1)]
basket_position = [list(map(int,input().split())) for _ in range(M)]

def reverse_ball(i,j) :
    reverse_basket = basket[i:j+1]
    reverse_basket.reverse()
    basket[i:j+1] = reverse_basket


for basket_nm in basket_position :
  i,j = basket_nm
  reverse_ball(i,j)

basket.pop(0)
basket = map(str,basket)
basket = " ".join(basket)
print(basket)